### ISSUE TITLE
Tab completion freezing ipython-console

### ISSUE THREAD

**Issue by nedclimaterisk** [ipython/ipython#11579]
(Note this is NOT about ctrl-space, I'm using <tab> to complete).

Sometimes when hitting autocomplete (<tab>), my console freezes - starts chewing CPU and becomes completely unresponsive. This is using ipython packages from conda/conda-forge:

```
python                    3.7.1                h5001a0f_0    conda-forge

ipykernel                 5.1.0           py37h24bf2e0_1001    conda-forge
ipython                   7.2.0           py37h24bf2e0_1000    conda-forge
ipython_genutils          0.2.0                      py_1    conda-forge
ipywidgets                7.4.2                      py_0    conda-forge
```
Is there any way that I could debug the behaviour? It's not obvious to me what's triggering it - tab-completion often seems to, but not always. Enough to be annoying.

This installation was copied from another laptop (same OS, kubuntu 18.04), and I have tried re-installing all ipython-related packages (those listed above). Is there anything else I should try reinstalling?

**Comment by Carreau** [comment]
probably not frozen, but just taking a long time the first time Jedi index the code for a given module. 
Do you have any specific example of which function trigger than and which one(s) do not ? 

Try to set [use_jedi](https://ipython.readthedocs.io/en/stable/config/options/terminal.html#configtrait-Completer.use_jedi) to false and then re-try.

**Comment by dashesy** [comment]
Even with turning off `use_jedi` Tab completion completely hangs, and I have to kill ipython

```
Python 3.6.8 |Anaconda, Inc.| (default, Dec 30 2018, 01:22:34) 
IPython 7.5.0 -- An enhanced Interactive Python. Type '?' for help.
```

CPU is at 100% forever

**Comment by ztaylor96** [comment]
Also having this issue. Did you find a solution?

**Comment by dashesy** [comment]
@ztaylor96 my issue was a 3rd party library infinite loop in init module. Even `Ctrl+C` did not work (and I had to kill ipython), which was the actual issue. So if as a result of Tab completion there is a bug in a library that goes into infinite loop `Ctrl+C` will not stop it, perhaps the indexer hides the signal.

**Comment by ztaylor96** [comment]
Interesting, thanks @dashesy.

**Comment by webbuildermn** [comment]
Thanks for this info. It happened to me in Condo on windows while taking a numpy course. Good to know I'm not insane. Sorry to learn there's no fix.

**Comment by javadba** [comment]
This froze in trying to use the `h20 datatable`.  How to set use_jedi to false?

**Comment by javadba** [comment]
It does generate the auto assist for pandas but freezes after that.

**Comment by javadba** [comment]
I put `use_jedi=False` in the REPL and that seemed to fixed pandas. Database is still hanging: but is it *really* indexing? There is no cpu usage from ipython.

**Comment by javadba** [comment]
This issue also occurs in _conda_ with _numpy_ on _macOS_

**Comment by barseghyanartur** [comment]
**TL;DR**

I downgraded my ipython to 7.1.0. That fixed it for me. I didn't try other versions.

**Details**

*Before (NOT working)*

ipython==7.16.1
prompt-toolkit==3.0.8

*After (DOES work)*

ipython==7.1.0
prompt-toolkit==2.0.10

**Comment by CervEdin** [comment]
It's not necessary to kill the `ipython` process, you can send a `kill -s signit` from another terminal to get out of the frozen state

**Comment by javadba** [comment]
problem still exists in ipython 7.27.0  with python 3.9.7

**Comment by matan1008** [comment]
I experience the same (in my humble opinion) problem.
Python 3.10.0
IPython 8.0.0
macOS 11.3
Can it be solved somehow?

**Comment by tdltdc** [comment]
This still appears to be the case.

Debian GNU/Linux trixie/sid
Python 3.11.7
IPython 8.14.0 (from the Debian Sid repository)

`pkill ipython3` unfreezes the console.
